src/DataViews/LiveFunctionsDataView.cpp

```diff
@@ -94,15 +94,16 @@ std::string LiveFunctionsDataView::GetValue(int row, int column) {
     return "";
   }
 
-  const uint64_t function_id = GetInstrumentedFunctionId(row);
-  const ScopeStats& stats = app_->GetCaptureData().GetScopeStatsOrDefault(function_id);
+  const uint64_t scope_id = GetScopeId(row);
+  const ScopeStats& stats = app_->GetCaptureData().GetScopeStatsOrDefault(scope_id);
 
-  const FunctionInfo& function = *GetFunctionInfoFromRow(row);
+  const FunctionInfo* function = GetFunctionInfoFromRow(row);
```

Comment:
So you changed GetFunctionInfoFromRow to return a pointer, to express it being null on manual instrumentation, right? I'd prefer to make it even more explicit and use std::optional<FunctionInfo> here instead.

Reply:
You're right. I've given std::optional a thought. Would it be `std::optional<FunctionInfo>`? This would copy an instance of `FunctionInfo`. Or rather `std::optional<FunctionInfo&>`, that would effectively say "the reference is a non-null pointer, but we wrap it into optional, so it can be null", thus making its semantics extremely close to that of a pointer. I don't see the benefit as long as we treat std::optional in the same way we treat pointers.

Reply:
There is similar discussions in go/totw/163.
I agree that `optional` here would be more readable, but the suggestion is to use pointer and let nullptr indicate "does not exist." 

I prefer to add a comments to the `GetFunctionInfoFromRow` method to explain when we return `nullptr` for this case.    
Previously `indices_` stores function ids for `FunctionInfo`s in `functions_`, this helps us to map the index / row in the view to the actual associated function.   
Now we change `indices_` to store scope id and it does not always have a matching `FunctionInfo` in the `functions_` if it a manually instrumented function.   
I would also like to add a comment above `indices_` and `functions_` to explain this a little bit.

Reply:
I don't think that copying FunctionInfo here is really a big deal, which is why I would prefer the optional.
However, I agree that we should not use optional<FunctionInfo&>, and rather use the nullptr with a comment version then.